### PR TITLE
[2.x] Add `toBeUuid` expectation

### PR DESCRIPTION
**Pull request by hungthai1401**
### What:

- [ ] Bug Fix
- [x] New Feature

### Description:

This PR adds `toBeUUID` expectation.

Usage:

```php
expect('00112233-4455-8677-8899-aabbccddeeff')->toBeUUID();
```

### CODE REVIEW

**Review by devajmeireles — APPROVED** [review]
Thanks for your PR.

**Comment by nexxai** [comment]
I have a decent amount of experience with UUIDs and this all looks decent to me.  I can't think of any edge cases that this regex wouldn't catch off the top of my head.

File: src/Mixins/Expectation.php [hunk 1]
```diff
@@ -1111,4 +1112,20 @@ public function toBeStudlyCase(string $message = ''): self
 
         return $this;
     }
+
+    /**
+     * Asserts that the value is UUID.
+     *
+     * @return self<TValue>
+     */
+    public function toBeUUID(string $message = ''): self
```
Comment (nunomaduro) [inline src/Mixins/Expectation.php]:
```suggestion
    public function toBeUuid(string $message = ''): self
```

**Review by nunomaduro — CHANGES_REQUESTED** [review]

**Comment by inxilpro** [comment]
I believe a slightly better regex would be:

`/^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/iD`

This accounts for known versions (1–5) and variants (8, 9, a, or b), while still allowing the NIL UUID. 

This pattern comes from [uuidjs](https://github.com/uuidjs/uuid)

**Comment by hungthai1401** [comment]
@inxilpro I think the current regex is good. This regex pattern comes from [laravel](https://github.com/laravel/framework/blob/52b301797d7a229bf7ebd3e2eefc762487738578/src/Illuminate/Support/Str.php#L519)

**Comment by gabbanaesteban** [comment]
There are different versions of UUIDs.

I would suggest using the approach used by https://www.npmjs.com/package/validator

Here is the validation https://github.com/validatorjs/validator.js/blob/master/src/lib/isUUID.js

**Review by nunomaduro — APPROVED** [review]

**Comment by nunomaduro** [comment]
@nexxai Can you PR docs?

**Comment by hungthai1401** [comment]
@nunomaduro I've created the documentation related to that: https://github.com/pestphp/docs/pull/229

**Pull request merged**